sql/core/src/main/scala/org/apache/spark/sql/execution/Columnar.scala

```diff
@@ -490,8 +490,10 @@ case class ApplyColumnarRulesAndInsertTransitions(conf: SQLConf, columnarRules:
       // The tree feels kind of backwards
       // Columnar Processing will start here, so transition from row to columnar
       RowToColumnarExec(insertTransitions(plan))
-    } else {
+    } else if (!plan.isInstanceOf[RowToColumnarExec]) {
```

Comment:
I had gone down the same path at one point with similar changes, but then had to introduce a new `RowToColumnarExecLike` trait so that it would also work with the `GpuRowToColumnarExec` implementation that we have in our plugin.

Reply:
do you mean https://github.com/apache/spark/pull/29067/files#diff-3a92abd70b28650055b16d8a101a95c3R55 ?

I can wait for it to be merged, and add something similar to `RowToColumnarExec`.

Reply:
I wonder if there is a more generic way to test for this, along the lines of `else if (plan.supportsColumnar && !plan.children.head.supportsColumnar)`?

Reply:
Yes, exactly like that.